index.html

```
@@ -2828,46 +2828,43 @@ <h3><dfn>Writing content</dfn></h3>
         or writing <a>NDEF messages</a>.
       </p>
       <p>
-        To <dfn>validate external type</dfn> given |input:USVString|, run
+        To <dfn>split external type</dfn> given |input:USVString|, run
         these steps:
       </p>
-      <ol class=algorithm id="validate-external-type">
-        <li>
-          If |input:string| is not an [=ASCII string=], is empty, or its
-          length exceeds 255 bytes, return `false`.
-        </li>
+      <ol class=algorithm id="split-external-type">
```

Shall we fail early when `:` is not found with something like this?
> 1.  If `U+003A` (`:`) is not found in |input|, return failure.
> 2. Let |domain| be the |input| from the start of |input| up to but excluding the first occurrence of `U+003A` (`:`).
> 3. Let |type| be the |input| after the first occurrence of `U+003A` (`:`) up to the end of |input|.
> 4. Return the |domain| and |type| pair.

---

Yeah makes sense

---

I don't know. To me it sounds more intuitive to have step 2 first since that is the real operation, and then step 1 as an error step of the former, otherwise someone might think they have to scan the string separately to check for `:`? 
No strong opinion here.

---

People can implement it that way as the result is the same, this is about clear spec text :-)